src/Calendar/components/TimeGrid/utils/useElementWidths.js

```diff
@@ -12,28 +12,33 @@ const useElementWidths = props => {
   const TimeGridRef = useRef(null);
   const [elementWidths, setElementWidths] = useState([]);
 
-  useEffect(() => {
-    const getElementsMeasurements = () => {
-      const widths = [];
-      elementRefs.forEach((element, day) => {
-        widths.push(element.offsetWidth);
-      });
-      return widths;
-    };
+  const getElementsMeasurements = () => {
+    const widths = [];
+    elementRefs.forEach((element, day) => {
+      widths.push(element.offsetWidth);
+    });
+    return widths;
+  };
 
-    const setAllWidths = () => {
-      if (!isEqual(getElementsMeasurements(), elementWidths)) {
-        setElementWidths(getElementsMeasurements());
-      }
-    };
+  const setAllWidths = () => {
+    if (!isEqual(getElementsMeasurements(), elementWidths)) {
+      setElementWidths(getElementsMeasurements());
+    }
+  };
 
-    const resizable = throttle(() => setAllWidths(), 300);
-    const ref = TimeGridRef.current;
+  const resizable = throttle(() => {
+    setAllWidths();
+  }, 300);
 
-    if (ref) addListener(ref, resizable);
+  useEffect(() => {
+    if (TimeGridRef.current) {
+      addListener(TimeGridRef.current, resizable);
+    }
     setAllWidths();
-    return () => removeListener(ref, resizable);
-  }, [elementRefs, elementWidths]);
+    return () => {
+      removeListener(TimeGridRef.current, resizable);
+    };
+  }, [resizable]);
```

Comment:
Ok confirmed it was this. This needs to re-render quite a bit.

Reply:
I think we could inline this, but we need to remove the listener if it's unmounted which is why useEffect here works